web/app/js/components/BaseTable.jsx

```diff
@@ -19,59 +21,125 @@ const styles = theme => ({
   table: {},
 });
 
-function BaseTable(props) {
-  const { classes, tableRows, tableColumns, tableClassName, rowKey, padding} = props;
+class BaseTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      order: this.props.defaultOrder || "asc",
+      orderBy: this.props.defaultOrderBy
+    };
+  }
 
-  return (
-    <Paper className={classes.root}>
-      <Table className={`${classes.table} ${tableClassName}`} padding={padding}>
-        <TableHead>
-          <TableRow>
-            { _.map(tableColumns, c => (
-              <TableCell
-                key={c.key}
-                numeric={c.isNumeric}>{c.title}
-              </TableCell>
-            ))
-            }
-          </TableRow>
-        </TableHead>
-        <TableBody>
-          {
-            _.map(tableRows, d => {
-            let key = !rowKey ? d.key : rowKey(d);
-            return (
-              <TableRow key={key}>
-                { _.map(tableColumns, c => (
-                  <TableCell
-                    key={`table-${key}-${c.key}`}
-                    numeric={c.isNumeric}>{c.render(d)}
-                  </TableCell>
-                  ))
-                }
-              </TableRow>
-            );
-          })}
-        </TableBody>
-      </Table>
-    </Paper>
-  );
+  createSortHandler = dataIndex => () => {
+    let orderBy = dataIndex;
+    let order = 'asc';
+
+    if (this.state.orderBy === orderBy && this.state.order === order) {
+      order = 'desc';
+    }
+
+    this.setState({ order, orderBy });
+  };
+
+  sortRows = (tableRows, tableColumns, order, orderBy) => {
+    if (!orderBy) {
+      return tableRows;
+    }
+
+    let col = _.find(tableColumns, ['dataIndex', orderBy]);
+    let sorted = tableRows.sort(col.sorter);
+    return order === 'desc' ? _.reverse(sorted) : sorted;
+  }
+
+  renderHeaderCell = (col, order, orderBy) => {
+    if (col.sorter) {
+      return (
+        <TableCell
+          key={col.key || col.dataIndex}
+          numeric={col.isNumeric}
+          sortDirection={orderBy === col.dataIndex ? order : false}>
+          <Tooltip
```

Comment:
I feel like an updown icon that's present all the time is a better visual indicator of whether a column is Sortable than the tooltip, since it's more discoverable. I'm not sure the tooltip gives us that much here.

<img width="749" alt="screen shot 2018-10-25 at 4 45 18 pm" src="https://user-images.githubusercontent.com/549258/47536524-63af3480-d875-11e8-9c84-08b0bcd32a42.png">

I couldn't find one on the default material icons page, I did find this one under "sort":
https://material.io/tools/icons/?icon=sort&style=baseline

Reply:
Oooh I found https://material.io/tools/icons/?icon=unfold_more&style=baseline